clusterctl/validation/validate_cluster_api_objects.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+func ValidateClusterAPIObjects(client *clientset.Clientset, clusterName string, namespace string) error {
+	fmt.Printf("Validating cluster API objects in namespace %q.\n", namespace)
+
+	cluster, err := getClusterObject(client, clusterName, namespace)
+	if err != nil {
+		return err
+	}
+	if err := validateClusterObject(cluster); err != nil {
+		return err
+	}
+
+	machines, err := client.ClusterV1alpha1().Machines(namespace).List(meta_v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to get the machines from the apiserver in namespace %q: %v", namespace, err)
+	}
+	if err = validateMachineObjects(machines); err != nil {
+		return err
+	}
+
```

Comment:
please create an TODO issue and link it here for the missing verification of MachineDeployments and MachineSets.